sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsTableRow.java

```diff
@@ -46,12 +54,69 @@ public int getRowIndex() {
      * Returns the value associated with the given column name. If the column name is not found
      * {@link Optional#isPresent()} evaluates to {@code false}.
      * @param columnName the column name for which the value is returned.
-     *
      * @return The value associated with the given column name.
      */
     public Optional<LogsTableCell> getColumnValue(String columnName) {
         return tableRow.stream()
-            .filter(cell -> cell.getColumnName().equals(columnName))
-            .findFirst();
+                .filter(cell -> cell.getColumnName().equals(columnName))
+                .findFirst();
+    }
+
+    /**
+     * Returns the table row as an object of the given {@code type}. The field names on the type should be
+     * reflectively accessible and the names should match the column names. If the table row contains columns that
+     * are not available on the object, they are ignored and similarly if the fields in the object are not found in
+     * the table columns, they will be null.
+     * @param type The type of the object to be returned
+     * @param <T> The class type.
+     * @return The object that this table row is mapped to.
+     * @throws IllegalArgumentException if an instance of the object cannot be created.
+     */
+    public <T> T getRowAsObject(Class<T> type) {
```

Comment:
Do we have any way of collecting usage\metric patterns? Since this method uses reflection - it will be slower than the other methods and so it might be worth measuring how popular this method is for our customers and do people use it because it is available vs because it is needed?